lib/Audio/Scan.pm

```diff
@@ -2,21 +2,8 @@ package Audio::Scan;
 
 use strict;
 
-our $VERSION;
-
-require XSLoader;
-
-BEGIN {
-	foreach ('0.99', '0.93', '0.95', '0.94') {
-		eval { XSLoader::load('Audio::Scan', $_); };
-		
-		if (!$@) {
-			$VERSION = $_;
-			last;
-		}
-	}
-}
-
+use XSLoader;
```

Comment:
If removing the version number, how would the comparison still work?

Reply:
$Audio::Scan::VERSION is set in the Audio::Scan module.

package Audio::Scan;
use strict;
our $VERSION = '1.01';

require XSLoader;
XSLoader::load('Audio::Scan', $VERSION);

The version works fine in the web gui information tab without it.

our $VERSION is lib/Audio/Scan.pm is never used after set.  there's no reference to it outside the code I removed.

-our $VERSION;
-
-require XSLoader;
-
-BEGIN {
-       foreach ('0.99', '0.93', '0.95', '0.94') {
-               eval { XSLoader::load('Audio::Scan', $_); };
-               
-               if (!$@) {
-                       $VERSION = $_;
-                       last;
-               }
-       }
-}
-

Reply:
But this _is_ the `Audio::Scan` module!

Reply:
I've been using these changes successfully for a while now. Perhaps I don't understand the relationship between lib/Audio/Scan.pm and the actual module which is in CPAN/arch/5.24/arm-linux-gnueabi-thread-multi-64int/Audio/Scan.pm on my test system, which has this at the beginning setting VERSION.

package Audio::Scan;
use strict;

our $VERSION = '1.01';

require XSLoader;
XSLoader::load('Audio::Scan', $VERSION);

Reply:
Changed Slim/Formats.pm to use slim::utils::versions

Reply:
Ah, ok: if you're installing your own Audio::Scan module inside the `arch` tree, then you're good. `lib/Audio/Scan.pm` is what is being used by the modules shipped by LMS. Yes, I've started using custom copies per platform etc. to simplify this, too. In your case you're fine. But by removing the version information from `lib/Audio/Scan.pm` you'd break most installations.

Reply:
ok.  I'll test on a stock lms install and refactor.  Thanks for your help.

Reply:
I've been unable to determine the version of Audio Scan.so programmatically during loading as there's no version attached to the C code shared module, only in Scan.pm.
One option would be to include versions 0.93 through 1.01 for Audio::Scan in the version loop or even higher for future verisons in lib/Scan/Audio.pm
The critical fix from the pull request is the change to Slim/Formats.pm for DSD based on the version of Audio::Scan.
I can revert the changes to modules.conf and lib/Auiod/Scan.pm or delete this PR and create a new one just for the Slim/Formats.pm change.

Reply:
Don't worry about this file. If you include "your" `Scan.pm` file with the binary int he platform specific folder, then this copy is ignored. The other one takes precedence. I started doing this a while ago, though it causes to carry around multiple copies of the same file. But with the file sizes we've reached by now I don't care about a few kB any more...

Just don't touch this file, leave it alone, and we'll be good :-)